ci/test_cpp.sh

```diff
@@ -32,9 +32,9 @@ trap "EXITCODE=1" ERR
 set +e
 
 rapids-logger "Running googletests"
-for gt in "$CONDA_PREFIX/bin/gtests/librmm/"* ; do
-    ${gt} --gtest_output=xml:${RAPIDS_TESTS_DIR}/
-done
+cd $CONDA_PREFIX/bin/gtests/librmm/
+export GTEST_OUTPUT=xml:${RAPIDS_TESTS_DIR}/
+ctest -j10 -VV
```

Comment:
@ajschmidt8 pointed out that highly verbose test logs [can pose a problem for the GHA GUI](https://github.com/rapidsai/ops/issues/2575). The current verbosity is an artifact of the looping approach IMO. Could we instead use `ctest -j10 --output-on-failure`?

Reply:
I was using very verbose to match the previous behavior. Very happy to switch over to ``--output-on-failure``

Reply:
Updated to use `--output-on-failure`